### ISSUE TITLE
evaluation error:double free or corruption

### ISSUE THREAD

**Issue by AddASecond** [balancap/SSD-Tensorflow#61]
**when I follow the evaluation process in Read.me, I obtain the error of:** 

Error in `python3.4': double free or corruption (!prev): 0x00000000016b7670 ***

**the code in my command file ‘eval.sh’ is:**
`DATASET_DIR=./VOC2007/test/
EVAL_DIR=./logs/  
CHECKPOINT_PATH=./checkpoints/ssd_300_vgg.ckpt
python3.4 eval_ssd_network.py \
    --eval_dir=${EVAL_DIR} \
    --dataset_dir=${DATASET_DIR} \
    --dataset_name=pascalvoc_2007 \
    --model_name=ssd_300_vgg \
    --checkpoint_path=${CHECKPOINT_PATH} \
    --batch_size=1
`

**the whole log from the terminal is shown as follows：**
I tensorflow/stream_executor/dso_loader.cc:135] successfully opened CUDA library libcublas.so.8.0 locally
I tensorflow/stream_executor/dso_loader.cc:135] successfully opened CUDA library libcudnn.so.5 locally
I tensorflow/stream_executor/dso_loader.cc:135] successfully opened CUDA library libcufft.so.8.0 locally
I tensorflow/stream_executor/dso_loader.cc:135] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:135] successfully opened CUDA library libcurand.so.8.0 locally
./VOC2007/test/

# =========================================================================== #
# Training | Evaluation flags:
# =========================================================================== #
{'batch_size': 1,
 'checkpoint_path': './checkpoints/ssd_300_vgg.ckpt',
 'dataset_dir': './VOC2007/test/',
 'dataset_name': 'pascalvoc_2007',
 'dataset_split_name': 'test',
 'eval_dir': './logs/',
 'eval_image_size': None,
 'eval_resize': 4,
 'gpu_memory_fraction': 0.1,
 'keep_top_k': 200,
 'master': '',
 'matching_threshold': 0.5,
 'max_num_batches': None,
 'model_name': 'ssd_300_vgg',
 'moving_average_decay': None,
 'nms_threshold': 0.45,
 'num_classes': 21,
 'num_preprocessing_threads': 4,
 'preprocessing_name': None,
 'remove_difficult': True,
 'select_threshold': 0.01,
 'select_top_k': 400,
 'wait_for_checkpoints': False}

# =========================================================================== #
# SSD net parameters:
# =========================================================================== #
{'anchor_offset': 0.5,
 'anchor_ratios': [[2, 0.5],
                   [2, 0.5, 3, 0.3333333333333333],
                   [2, 0.5, 3, 0.3333333333333333],
                   [2, 0.5, 3, 0.3333333333333333],
                   [2, 0.5],
                   [2, 0.5]],
 'anchor_size_bounds': [0.15, 0.9],
 'anchor_sizes': [(21.0, 45.0),
                  (45.0, 99.0),
                  (99.0, 153.0),
                  (153.0, 207.0),
                  (207.0, 261.0),
                  (261.0, 315.0)],
 'anchor_steps': [8, 16, 32, 64, 100, 300],
 'feat_layers': ['block4', 'block7', 'block8', 'block9', 'block10', 'block11'],
 'feat_shapes': [(38, 38), (19, 19), (10, 10), (5, 5), (3, 3), (1, 1)],
 'img_shape': (300, 300),
 'no_annotation_label': 21,
 'normalizations': [20, -1, -1, -1, -1, -1],
 'num_classes': 21,
 'prior_scaling': [0.1, 0.1, 0.2, 0.2]}

#=========================================================================== #
# Training | Evaluation dataset files:
# =========================================================================== #
*** Error in `python3.4': double free or corruption (!prev): 0x00000000016b7670 ***
./evaluation0412.sh: line 11: 38357 Aborted                 (core dumped) python3.4 eval_ssd_network.py --eval_dir=${EVAL_DIR} --dataset_dir=${DATASET_DIR} --dataset_name=pascalvoc_2007 --model_name=ssd_300_vgg --checkpoint_path=${CHECKPOINT_PATH} --batch_size=1

**Comment by ljdang** [comment]
@bobauditore I got the same error. Did you fix it?

**Comment by AddASecond** [comment]
@ljdang Sadly to say that No I didn‘t.

**Comment by ljdang** [comment]
@bobauditore how much GPU memory with your computer? Maybe it is the reason.

**Comment by AddASecond** [comment]
@ljdang I don‘t think so because I use the server with 4 TITAN X(Pascal) devises.

**Comment by sleepfin** [comment]
@bobauditore Have you fixed the error? I am dealing with the "evaluation error:double free or corruption
" as well

**Comment by sleepfin** [comment]
@bobauditore @ljdang 
This works for me.
See https://github.com/tensorflow/tensorflow/issues/6968
If you are using Ubuntu, your command is:
```
sudo apt-get install libtcmalloc-minimal4
export LD_PRELOAD="/usr/lib/libtcmalloc_minimal.so.4"
```
since I am using redhat, my cmd is:
```
sudo yum install gperftools-libs-2.4-8.el7.x86_64
export LD_PRELOAD="/usr/lib64/libtcmalloc_minimal.so.4"
```
